### PR TITLE
static-build: fix potential NULL dereference in openssl

### PR DESCRIPTION
**Pull request by CuriousGeorgiy**
Closes tarantool/security#27

### CODE REVIEW

File: static-build/cmake/AddDependencyProjects.cmake [hunk 1]
```diff
@@ -74,8 +74,10 @@ ExternalProject_Add(openssl
         --libdir=lib
         no-shared
     INSTALL_COMMAND ${CMAKE_MAKE_PROGRAM} install_sw
-    PATCH_COMMAND patch -d <SOURCE_DIR> -p1 <
-        "${PATCHES_DIR}/openssl-111q-gh-18720.patch"
+    PATCH_COMMAND cat
+    "${PATCHES_DIR}/openssl-111q-gh-18720.patch"
+    "${PATCHES_DIR}/openssl-tarantool-security-gh-27.patch" |
+    patch -d <SOURCE_DIR> -p1
```
Comment (locker) [inline static-build/cmake/AddDependencyProjects.cmake:80]: Nit: bad indentation.

**Review by locker — CHANGES_REQUESTED** [review]
1. Please rebase on the latest master (needed for `static-build-ci` label).
2. Please split the ticket by project so that there's one PR/patch per ticket and assign the part about openssl to yourself so that nobody tries to fix it again in the meantime.

File: static-build/cmake/AddDependencyProjects.cmake (hunk 1 above)
Comment (CuriousGeorgiy) [reply static-build/cmake/AddDependencyProjects.cmake]: Fixed.

**Comment by CuriousGeorgiy** [comment]
@locker 

1. Rebased the branch.
2. @gumix has split the tickets.

**Review by locker — CHANGES_REQUESTED** [review]

**Review by Gumix — APPROVED** [review]

**Review by locker — APPROVED** [review]

**Comment by locker** [comment]
Cherry-picked to 2.10.

**Pull request merged**